autopush/endpoint.py

```diff
@@ -95,148 +182,278 @@
+For GCM, this is the `SenderID` and is pre-negotiated outside of the push
+service.
+
+   :{instanceid}: The bridge specific private identifier token
+
+Each protocol requires a unique token that addresses the
+application on a given user's device. This is usually the product of the
+application registering the {instanceid} with the native bridge protocol
+agent.
+
+   :{auth_token}: The Authorization token
+
+Most calls to the HTTP interface require a Authorization header. The
+Authorization header is a bearer token, which has been provided by the
+**Registration** call and is preceded by the token type word "Bearer".
+
+An example of the Authorization header would be:
+
+::
 
-    :reqheader Authorization: Bearer ...
+    Authorization: Bearer 0123abcdef
 
-.. http:post:: /register/(uuid:uaid)
+Calls
+-----
 
-    Create a new endpoint for a given UAID, or register a new UAID and
-    return a new endpoint.
+Registration
+~~~~~~~~~~~~
+
+Request a new UAID registration, Channel ID, and optionally set a bridge
+type and token for this connection.
+
+Call:
+^^^^^
+
+.. http:post:: /v1/{type}/{token}/registration
+
+This call requires no Authorization for first time use.
+
+Parameters:
+^^^^^^^^^^^
+
+    {"token":{instanceid}}
 
     .. note::
 
-      If a ``channelID`` field is not supplied then one will be generated.
-      Supplied ``channelID`` fields must be valid UUID hex strings.
+        If additional information is required for the bridge, it may be
+        included in the paramters as JSON elements. Currently, no additional
+        information is required.
 
-    **1. Endpoint w/New Registration**
+Reply:
+^^^^^^
 
-    **Example Request (for APNS router)**:
+.. code-block:: json
 
-    .. code-block:: http
+    `{"uaid": {UAID}, "secret": {auth_token},
+    "endpoint": "https://updates-push...", "channelID": {CHID}}`
 
-        POST /register
-        Host: endpoint.push.com
-        Content-Type: application/json
+example:
 
-        {
-            "type": "apns",
-            "data": {
-                "token": "APNS_TOKEN_DATA",
-                ...
-            }
-        }
+.. code-block:: http
 
-    **Example Response**:
+    > POST /v1/gcm/a1b2c3/registration
+    >
+    > {"token": "1ab2c3"}
 
-    .. code-block:: http
+.. code-block:: json
 
-        HTTP/1.1 200 OK
-        Content-Type: application/json
-        Content-Length: nnn
+    < {"uaid": "abcdef012345",
+    < "secret": "0123abcdef",
+    < "endpoint": "https://updates-push.services.mozaws.net/push/...",
+    < "channelID": "01234abcd"}
 
-        {
-            "uaid": "5bbc4aae-a575-4f6a-a4b3-6f84a4d06a63",
-            "secret": "4526e381eb6c191cf783da5d6df248e7",
-            "channelID": "a13872c9-5cba-48ab-a8e5-955264647303",
-            "endpoint": "https://endpoint.push.com/push/VERYLONGSTRING",
-        }
+Errors:
+^^^^^^^
 
-    **2. Endpoint w/Existing UAID**
+Standard error codes apply
```

Comment:
nit: consistent strings. The ones above all had periods at the end, now it seems to switch to no periods